docs/backend/guides/how-to/create-a-custom-linter.md

```diff
@@ -0,0 +1,177 @@
+
+<!-- toc -->
+
+* [Setting up your linter](#setting-up-your-linter)
+  * [File structure](#file-structure)
+  * [Analyzer](#analyzer)
+  * [Creating a linter](#creating-a-linter)
+  * [Writing linter tests](#writing-linter-tests)
+* [Testing the linter across files](#testing-the-linter-across-files)
+  * [Utilizing test data](#utilizing-test-data)
+* [Running the linter in pre-commit](#running-the-linter-in-pre-commit)
+* [Additional Resources](#additional-resources)
+
+
+Regenerate with "pre-commit run -a markdown-toc"
+
+<!-- tocstop -->
+
+Creating custom GO linters can be a great way to analyze your 
+project's source code and alert you to bugs, errors, 
+or other issues with your code.
+
+## Setting up your linter
+
+### File structure
+Start by setting up the files you'll need for your linter:
+```golang
+-cmd
+    -example_linter_name
+        -main.go
+
+-pkg
+    -example_linter_name
+        -example_linter.go
+        -example_linter_test.go
+```
+In the `cmd` folder create a folder for your linter called `<example_linter_name>` and add an empty file called `main.go`
+
+In `pkg` folder create another folder for your linter called `<example_linter_name>` and in it you will 
+place your `example_linter.go` and `example_linter_test.go`
+
+### Analyzer:
+The `cmd` folder contains the analysis driver for your linter. The file `main.go` will utilize a package called `singlechecker`,
+which defines the main function for an analysis driver and provides a tool to run a standalone analysis.
+
+The code in `main.go` is:
+
+```golang
+package main
+
+import (
+	"golang.org/x/tools/go/analysis/singlechecker"
+
+	examplelinter "github.com/transcom/mymove/pkg/example-linter"
+)
+
+func main() { singlechecker.Main(examplelinter.LinterAnalyzer) }
+```
+
+### Creating a linter
+The linter will live in the `pkg` folder, in a folder named after your linter. 
+In `linter.go` you will store the linter analyzer that gets referenced in `cmd/linter_name/main.go`. 
+It will contain the name of the liner, documentation about the linter, a call to runthe linter, and requirements:
+
+```golang
+var LinterAnalyzer = &analysis.Analyzer{
+	Name:     "linternamelint",
+	Doc:      "Make sure X object is properly used throughout codebase",
+	Run:      run,
+	Requires: []*analysis.Analyzer{inspect.Analyzer},
+}
+```
+
+Note that `Run` expects an interface and runs your linter code. This value can be the function that executes your linter. In this example, we are calling the function `run`:
+
+```golang
+func run(pass *analysis.Pass) (interface{}, error) {
+    // Some code
+}
+```
+
+The linter is actually analyzing an abstract syntax tree, AST, that represents code in a file. 
+When your linter gets to a position in a file, where it catches an error, bug, or issue, it will flag this for the user.
+Because the linter is analyzing an AST, your code must be able to search through a file and mark which position it is when an issue is caught.
```

Comment:
I'm not sure if it would be good or not to include an example of what it looks like to mark a position with a message.

Reply:
What do you mean?

Reply:
Like showing an example like this: 
https://github.com/transcom/mymove/blob/d8d2b3862a28b344a1afdbb1a781d6529f04feb8/pkg/appcontext-linter/appctx.go#L46

To show that you call the `.Pos()` method on an AST object to get its position and pass that along with a message to `pass.Reportf` to report an issue found in the code. Maybe this is getting to specific though and they should just get that from reading the linked article on how to write the `run` function.